distributions/test/src/test/java/com/connexta/ion/replication/distributions/test/ReplicationITest.java

```diff
@@ -0,0 +1,144 @@
+
+  private static final Slf4jLogConsumer logConsumer = new Slf4jLogConsumer(LOGGER);
+
+  private Set<String> idsCreated = new HashSet<>();
+
+  @BeforeClass
+  public static void beforeClass() {
+    replication.followOutput(logConsumer);
+    solrUrl = "http://" + solr.getContainerIpAddress() + ":" + solr.getFirstMappedPort();
+    sourceUrl = System.getenv("source.node");
+    destinationUrl = System.getenv("destination.node");
+    if (sourceUrl == null) {
+      sourceUrl = "https://ditto-1.phx.connexta.com:8993/services";
+    }
+    if (destinationUrl == null) {
+      destinationUrl = "https://ditto-2.phx.connexta.com:8993/services";
+    }
+  }
+
+  @After
+  public void cleanUp() {
+    idsCreated.forEach(this::deleteFromNodes);
+    clearAllSolrCores();
+  }
+
+  @Test
+  public void createXmlFile() throws Exception {
+    LOGGER.info("Solr URL: {}", solrUrl);
+    String testId = UUID.randomUUID().toString();
+
+    // create product
+    String id =
+        createProductOnSourceNode(
+            APPLICATION_XML, "data/reuters-xml-sample", ImmutableMap.of("testId", testId));
+    LOGGER.info("Returned ID: {}", id);
+    idsCreated.add(id);
+
+    // create nodes and jobs
+    postToSolrCore(
+        SITE, "sites.json", ImmutableMap.of("source", sourceUrl, "destination", destinationUrl));
+    postToSolrCore(CONFIG, "jobs.json", ImmutableMap.of("testId", testId));
+
+    // wait for product to be replicated
+    await()
+        .with()
+        .pollInterval(1, TimeUnit.SECONDS)
+        .atMost(40, TimeUnit.SECONDS)
+        .until(queryByIdSucceeds(id));
+
+    // confirm metacard has resource
+    XmlPath xmlPath =
+        given()
+            .relaxedHTTPSValidation()
+            .when()
+            .get(destinationUrl + CATALOG_ENDPOINT + "/" + id)
+            .then()
+            .extract()
+            .body()
+            .xmlPath();
+
+    // TODO: should I verify the replication origins attribute is added?
+    // this will make it more difficult to make the source and destination configurable
+    assertTrue(
+        xmlPath
+            .getList("metacard.string.find { it.@name == 'metacard-tags' }.value", String.class)
+            .contains("replicated"));
+    assertTrue(
+        xmlPath
+            .getString("metacard.string.find { it.@name == 'title' }.value[0]")
+            .contains(testId));
+  }
+
+  private void postToSolrCore(
+      String core, String jsonFilePath, ImmutableMap<String, String> varReplacementMap) {
+    given()
+        .header(CONTENT_TYPE, APPLICATION_JSON)
+        .body(getFileAsText(jsonFilePath, varReplacementMap))
+        .when()
+        .post(solrCoreUpdateUrl(core))
+        .then()
+        .statusCode(200);
+  }
+
+  private void deleteFromNodes(String id) {
+    given().relaxedHTTPSValidation().when().delete(sourceUrl + CATALOG_ENDPOINT + "/" + id);
+    given().relaxedHTTPSValidation().when().delete(destinationUrl + CATALOG_ENDPOINT + "/" + id);
```

Comment:
❓ This cleanup might end up failing on the source and then the destination would not be cleaned up right?

Reply:
❓ Would it be easier to do a cleanup that simply cleans it all instead of trying to clean specific entries? Kind of like a reset? This would be a reusable test rule.

Reply:
We don't do any assertion here but yea if the source delete throws some kind of exception the destination might not get cleaned up. To delete everything would interfere with testing someone else might be performing with the test instances. To delete just what we created would require a delete by query which isn't possible as far as I know.

Reply:
This test case will have to change as it cannot rely on the pre-existence of another container by hardcoding external URLs as it currently does. I though these were temporary for the `WIP`. Tests must be self contained.

Reply:
not sure what you mean about hardcoding urls, you can set the source and destination urls using an environment variable but they default to ditto-1 and ditto-2. It turns out I can do a delete by query but then we have to wait for solr to perform a hard commit of what we're deleting. If we delete by id we don't have to wait, we can just remove it right away.

Reply:
My point on hardcoding is that the test expects an external system to be up and running. This is an external dependency that cannot be controlled by the test case and thus will render them brittle and unpredictable. The fact that the URL can be passed in or default to a hardcoded `ditto-1` is the proof of this issue. It might be acceptable for a very short time but it needs to be addressed somehow and very soon.

Reply:
got it for the delete by id. Just put some error handling in the method to not suddenly skip the destination if the source failed for whatever reasons.